src/main/java/com/burgerking/Menu.java

```diff
@@ -95,78 +110,50 @@ public void filterMenuByFoodType(int filterOption) {
         if(filterOption < 0 || filterOption > FoodType.values().length){
             throw new FoodTypeNotFoundException("There is no such food type in menu!");
         }
-        List<Food> filtered = new ArrayList<>();
-        for(Food food: this.getProductList()){
-            if(food.getType().ordinal()==filterOption){
-                filtered.add(food);
-            }
-        }
+        List<Food> filtered = this.productList.stream()
+                .filter(food -> food.getType().ordinal()==filterOption).collect(Collectors.toList());
         if(filtered.isEmpty()){
             throw new FilteredListIsEmptyException("No food of this type in menu - nothing to filter!");
         }
         this.setFilteredList(filtered);
     }
 
     public boolean isAnyFoodPriceBiggerThan(Double price) {
-        for (Food food : this.getProductList()) {
-            if (food.getPrice() > price) {
-                return true;
-            }
-        }
-        return false;
+        return this.productList.stream().anyMatch(food -> food.getPrice()>price);
     }
 
     public boolean isAllFoodCaloriesLowerThen(Integer calories){
-        for(Food food: this.getProductList()){
-            if(food.getCalories() > calories){
-                return false;
-            }
-        }
-        return true;
+        return this.productList.stream().allMatch(food -> food.getCalories()<calories);
     }
 
     public boolean isNoneFoodSizeEqualsTo(String size){
-        for(Food food: this.getProductList()){
-            if(food.getSize().equalsIgnoreCase(size)){
-                return false;
-            }
-        }
-        return true;
+        return this.productList.stream().noneMatch(food -> food.getSize().equalsIgnoreCase(size));
     }
 
-    public List<Food> getLowestPriceByType(){
+    public String getLowestPriceByType(){
         List<Food> typesList = new ArrayList<>();
-        List<Food> returnList = new ArrayList<>();
         for(int i=0;i<FoodType.values().length;i++){
-            for(Food food: this.getProductList()) {
-                if (food.getType().ordinal() == i) {
-                    typesList.add(food);
-                }
-            }
-            if (typesList.size()>0) {
-                typesList.sort(compareByPrice);
-                returnList.add(typesList.get(0));
-            }
-            typesList.clear();
+            int filterOption = i;
```

Comment:
you may iterate over FoodType.values() array using enhanced for, it will be a bit more simple and there will be no need in using ordinals

Reply:
i changed it to foreach loops that uses types - or what do you mean on "enhanced for"?